archiver.rb

```diff
@@ -0,0 +1,36 @@
+require 'tempfile'
+require 'fileutils'
+require 'find'
+
+class Archiver
+  attr_accessor :file
+  @@modes = [:reader, :writer, :both]
+
+  def initialize
+  end
+
+  @@modes.each do |mode|
+    define_method("open_file_as_#{mode}") do |file|
+      @file = File.open(file, 'r') if mode == :reader
+      @file = File.open(file, 'w') if mode == :writer
+      @file = File.open(file, 'r+') if mode == :both
+    end
+  end
+
+  def read_file lines: false
+    lines ? data = self.file&.read : data = self.file&.readlines.map(&:chomp)
```

Comment:
It is not needed to use self in order to get the value of the file, the attr_accessor is creating for you two methods "file" and "file=" (getter and setter) 
https://www.rubyguides.com/2018/11/attr_accessor/